src/mmw/js/src/modeling/models.js

```diff
@@ -298,6 +298,12 @@ var ScenarioModel = Backbone.Model.extend({
             App.currProject.saveProjectAndScenarios();
             return;
         }
+        if (this.isNew() && this.saveCalled) {
+            return;
+        } else if (this.isNew() && !this.saveCalled) {
+            // Makeshift locking mechanism to prevent double saves.
+            this.saveCalled = true;
```

Comment:
Since this flag is playing an important role, I think we should define it explicitly on the class like we do for `ProjectModel`, not just implicitly when we first set it.